<launch>

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />

    <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

    <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_bump_tolerant_control_manager" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <!-- Namespace - all topics, services and parameters will be remapped using this namespace as a prefix (eg.: "/bump_tolerant_control/odom_uav_in" to "/uav1/bump_tolerant_control/odom_uav_in") -->
  <group ns="$(arg UAV_NAME)">

    <!-- BumpTolerantControl nodelet -->
    <node pkg="nodelet" type="nodelet" name="bump_tolerant_controller" args="$(arg nodelet) bump_tolerant_controller_plugin/bump_tolerant_controller/BumpTolerantController $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <!-- Parameters loaded from launch file -->
      <param name="uav_name" value="$(arg UAV_NAME)" />

        <!-- ROS parameters config file -->
      <rosparam file="$(find bump_tolerant_control)/config/bump_tolerant_control.yaml" />

        <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->
        <remap from="~contact_point_in"   to="estimation_manager/contact_point" />
        <remap from="~external_wrench_in" to="estimation_manager/ext_wrench" />
        <remap from="~imu_in" to="hw_api/imu" />


        <!-- Publishers (note that the topic names will also be remapped using the namespace as a prefix) -->


    </node>

  </group>

</launch>
